### ISSUE TITLE
Needs tests

### ISSUE THREAD

**Issue by benwiley4000** [benwiley4000/react-responsive-audio-player#35]
We could use some component tests, as the API has become a bit complex to test just by loading example.html in a web browser.

**Comment by benwiley4000** [comment]
If you're here from Hacktoberfest you can also help out by [adding documentation](https://github.com/benwiley4000/cassette/issues/298)! 😃

**Comment by danielr18** [comment]
I think it's because `babel.config.js` was added on babel 7 beta 46, and we are currently at 41. I'll update babel 7 packages and push the changes if I get JSX working.

**Comment by benwiley4000** [comment]
Ah!! You're probably right, I forgot I was still on an old beta version.

**Comment by benwiley4000** [comment]
Jest stuff is now merged into the next branch. I've added one passing snapshot test for now, as a starting point. Obviously we want more coverage. :)

**Comment by benwiley4000** [comment]
This comment was originally posted [here](https://github.com/benwiley4000/cassette/issues/183#issuecomment-446765382) but makes more sense in the general testing issue:

---------

I think Jest is a good tool to use. I could use some help brainstorming, but here's a starter checklist:

## Component snapshot tests
### `@cassette/player`
Each of these should be wrapped in context providers with mock values (the actual Context.Provider, not the smart wrapper):
- [ ] `MediaPlayerControls`
- [ ] All the control components
- Stuff that might be hard to test:
    - `VolumeControl` - can we mock different internal states?
### `@cassette/components`
- [ ] `ProgressBarDisplay`
- [ ] Maybe `ProgressBar` (just to mock the `adjusting` state?)
- [ ] `MaybeMarquee`
- [ ] `VideoDisplay`
- Stuff that might be hard to test:
    - Potentially everything except for `ProgressBarDisplay`? These might be lower priority, especially since they're each so dependent on the DOM and on internal states.
## Unit tests for util functions
The various separated util functions could all have tests, and perhaps new util functions could be pulled out
## Context tests
It might require some refactoring and some mocks e.g. of the HTMLVideoElement, but it would be good to have API tests for the `playerContext` and `fullscreenContext`. I'm imagining that for the majority of functions available through the context, we have a test or tests that call the function then check afterward that:
1. The expected update occurred in the mock HTMLVideoElement
2. The expected change occurred in the context
Many of these updates will be asynchronous and some might even not be done on the next tick, so coding this could be tricky.

Mocking stuff from the browser will be an issue to resolve.

**Comment by benwiley4000** [comment]
These tests don't need to be done before the beta release is out. They can be part of the beta stabilization phase.
